python/packages/nisar/workflows/rubbersheet.py

```diff
@@ -22,9 +24,197 @@
+                    # Dense offset is enabled and offset products are disabled
+                    dense_offsets_path = pathlib.Path(rubbersheet_params['dense_offsets_path'])
+                    dense_offsets_dir = dense_offsets_path / 'dense_offsets' / f'freq{freq}' / pol
+                    corr_peak_path = str(f'{dense_offsets_dir}/correlation_peak')
+                    dense_offsets_path =  str(f'{dense_offsets_dir}/dense_offsets')
+                else:
+                    off_prod_dir = scratch_path / 'offsets_product' / f'freq{freq}' / pol
+                    # Get layer keys
+                    layer_keys = [key for key in
+                                  cfg['processing']['offsets_product'].keys() if
+                                  key.startswith('layer')]
+
+                    # Get correlation peak path for the last offset layer since it has larger window size
+                    corr_peak_path = str(f'{off_prod_dir}/{layer_keys[-1]}/correlation_peak')
+                    dense_offsets_path =  str(f'{off_prod_dir}/{layer_keys[-1]}/dense_offsets')
+
+                # Sampling
+                az_sampled_indices = np.linspace(0,
+                                                 len(off_az_indices)-1,
+                                                 rubbersheet_params['polyfitting']['samples_along_azimuth'],
+                                                 dtype=int)
+                rg_sampled_indices = np.linspace(0,
+                                                 len(off_rg_indices)-1,
+                                                 rubbersheet_params['polyfitting']['samples_along_range'],
+                                                 dtype=int)
+
+                az_polyfit_indices = off_az_indices[az_sampled_indices]
+                rg_polyfit_indices = off_rg_indices[rg_sampled_indices]
+
+                lines, pixels = np.meshgrid(az_sampled_indices,
+                                            rg_sampled_indices,
+                                            indexing="ij")
+
+                corr_peak = _open_raster(corr_peak_path, 1)
+                az_offsets = _open_raster(dense_offsets_path, 1)
+                rg_offsets = _open_raster(dense_offsets_path, 2)
+
+                corr_peak = corr_peak[lines, pixels]
+                az_offsets = az_offsets[lines, pixels]
+                rg_offsets = rg_offsets[lines, pixels]
+
+                data = np.column_stack([
+                    np.arange(corr_peak.size),
+                    np.repeat(az_polyfit_indices, len(rg_polyfit_indices)),
+                    np.tile(rg_polyfit_indices, len(az_polyfit_indices)),
+                    az_offsets.ravel(), rg_offsets.ravel(),
+                    corr_peak.ravel()
+                ])
+
+                # polyfitting the range and azimuth offsets
+                results = offsets_polyfit.polyfit_offsets(
+                    data,degree=rubbersheet_params['polyfitting']['degree'],
```

Comment:
```suggestion
                    data,
                    degree=rubbersheet_params['polyfitting']['degree'],
```
Original construction looked as if the value at `rubbersheet_params['polyfitting']['degree']` was being split as a tuple and assigned to the keywords `data` and `degree`